File: .docker/server_http/www/tests/python/controlCode.py
import random

da1 = ""

# fonction obligatoire
def stdin():
    global da1
    a1 = []
    a3 = 'P,C,T,K-1,2,3,4,5,6,7,8,9,10,V,D,R'.split("-")
    a2 = a3[0].split(",")
    s2 = a3[1].split(",")
    for s1 in a2:
        for s3 in s2:
            a1.append(s3 + s1)
    nr = random.randint(0, len(a1)-1)
    da1 = a1[nr]
    del a1[nr]
    random.shuffle(a1)
    return a1

currentBuffer = ""

def stdout(data):
    global currentBuffer
    currentBuffer += data

# fonction obligatoire
def hoTryCode():
    global da1
    try:
        returnCard = main()
        if returnCard == da1:
            print(True)
        else:
            print(f"Votre code a retourne {returnCard}, la carte manquante etait {da1}")
    except Exception as e:
        print("Il y a une erreur dans votre code\n")

File: .docker/server_http/www/tests/php/controlCode.php
<?php
$da1;
// fonction obligatoire
function stdin() {    
    global $da1;
    $a1=[];$a3=explode('-','P,C,T,K-1,2,3,4,5,6,7,8,9,10,V,D,R');
    $a2=explode(',',$a3[0]);$s2=explode(',',$a3[1]);
    foreach((array)$a2 as $s1){foreach((array)$s2 as $s3){$a1[]=$s3.$s1;}}
    $nr=rand(0,count($a1)-1);
    $da1=$a1[$nr];
    unset($a1[$nr]);
    shuffle($a1);

    return $a1;
}

// fonction obligatoire
function hoTryCode() {
    global $a1, $da1;
    if (function_exists('main')) {
        try {
            $returnCard = main();
            if ($returnCard === $da1) {
                return true;
            } else {
                return "Votre code a retourné $returnCard, la carte manquante était $da1";
            }
        } catch (Exception $e) {
            return "Il y a une erreur dans votre code : " . $e->getMessage();
        }
    } else {
        return 'Error function main not found';
    }
}